### PR TITLE
Add collect referenced nuget packages target

### PR DESCRIPTION
**Pull request by jmarolf**
adding CollectReferencedNuGetPackages so consumers that partially import VS targets are not broken.

VS Bug [526825](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems/edit/526825)

### CODE REVIEW

File: src/Tasks/Microsoft.Common.CurrentVersion.targets [hunk 1]
```diff
@@ -5656,6 +5656,10 @@ Copyright (C) Microsoft Corporation. All rights reserved.
   <!-- Get the resolved SDK reference items -->
   <Target Name="GetResolvedSDKReferences" DependsOnTargets="ResolveSDKReferences" Returns="@(ResolvedSDKReference)"/>
 
+  <!-- Target will return no results by default but will be overridden by other targets elsewhere.
```
Comment (Pilchie) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets]: I don't think this comment is technically accurate.  This target isn't overridden anywhere, it's just that other targets may happen to populate `ReferencedNuGetPackages` as a side-effect, and we want to collect them as a return value if they do.

**Comment by AndyGerlicher** [comment]
@dotnet-bot test this please

**Comment by AndyGerlicher** [comment]
@dotnet-bot test this please

**Comment by AndyGerlicher** [comment]
Per email thread changing to master for 15.6 (FYI I had to rebase and push your branch).

**Pull request merged**